lm/srilm.py

```diff
@@ -0,0 +1,548 @@
+        mem_rqmt: int = 48,
+        time_rqmt: float = 24,
+        cpu_rqmt: int = 1,
+        fs_rqmt: str = "100G",
+    ):
+        """
+        :param ngram_order: Maximum n gram order
+        :param data: Input data to be read as textfile
+        :param extra_count_args: Extra arguments for the execution call e.g. ['-unk']
+        :param count_exe: Path to srilm ngram-count executable
+        :param mem_rqmt: Memory requirements of Job (not hashed)
+        :param time_rqmt: Time requirements of Job (not hashed)
+        :param cpu_rqmt: Amount of Cpus required for Job (not hashed)
+        :param fs_rqmt: Space on fileserver required for Job, example: "200G" (not hashed)
+
+        Example options/parameters for count_args:
+        -unk
+        """
+        self.ngram_order = ngram_order
+        self.data = data
+        self.count_args = extra_count_args if extra_count_args is not None else ["-unk"]
+        self.count_exe = get_ngram_count_exe(count_exe)
+
+        self.out_counts = self.output_path("counts", cached=True)
+
+        self.rqmt = {
+            "mem": mem_rqmt,
+            "time": time_rqmt,
+            "cpu": cpu_rqmt,
+            "qsub_args": f"-l h_fsize={fs_rqmt}",
+        }
+
+    def tasks(self):
+        yield Task("create_files", mini_task=True)
+        yield Task("run", resume="run", rqmt=self.rqmt)
+
+    def create_files(self):
+        """creates bash script that will be executed in the run Task"""
+        cmd = [
+            f"{self.count_exe} \\\n",
+            f"  -text {self.data.get_path()} \\\n",
+            f"  -order {self.ngram_order} \\\n",
+            f"  -write counts \\\n",
+            f"  {' '.join(self.count_args)} -memuse\n",
+        ]
+
+        create_executable("run.sh", cmd)
+
+    def run(self):
+        """executes the previously created bash script and relinks outputs from work folder to output folder"""
+        self.sh("./run.sh")
+        relink("counts", self.out_counts.get_path())
+
+    @classmethod
+    def hash(cls, kwargs):
+        """delete the queue requirements from the hashing"""
+        del kwargs["mem_rqmt"]
+        del kwargs["cpu_rqmt"]
+        del kwargs["time_rqmt"]
+        del kwargs["fs_rqmt"]
+        return super().hash(kwargs)
+
+
+class ComputeNgramLmJob(Job):
+    """
+    Generate count based LM with SRILM
+    """
+
+    class DataMode(Enum):
+        TEXT = 1
+        COUNT = 2
+
+    def __init__(
+        self,
+        ngram_order: int,
+        data: tk.Path,
+        data_mode: DataMode,
+        vocab: Optional[tk.Path] = None,
+        extra_ngram_args: Optional[List[str]] = None,
+        count_exe: Optional[tk.Path] = None,
+        mem_rqmt: int = 48,
+        time_rqmt: float = 24,
+        cpu_rqmt: int = 1,
+        fs_rqmt: str = "100G",
+    ):
+        """
+        :param ngram_order: Maximum n gram order
+        :param data: Either text file or counts file to read from, set data mode accordingly
```

Comment:
Can the count file come from the `CountNgramsJob`s `out_counts`? If yes I think this would be nice to add here.